app/uk/gov/hmrc/tai/util/TaxYearRangeUtil.scala

```diff
@@ -50,17 +52,31 @@ object TaxYearRangeUtil {
     messages("tai.taxYear", start, end)
   }
 
-  private def dateRangeSingleLine(messageKey: String, from: LocalDate, to: LocalDate)(
-    implicit messages: Messages): String =
-    HtmlFormatter.htmlNonBroken(dateRange(messageKey, from, to))
+  private def createDateFormatForPattern(pattern: String)(implicit messages: Messages): SimpleDateFormat = {
+    val uLocale = new ULocale(messages.lang.code)
+    val validLang: Boolean = ULocale.getAvailableLocales.contains(uLocale)
+    val locale: ULocale = if (validLang) uLocale else ULocale.getDefault
+    val sdf = new SimpleDateFormat(pattern, locale)
+    sdf.setTimeZone(TimeZone.getTimeZone("Europe/London"))
+    sdf
+  }
+
+  private def jodaDate2javaDate(date: LocalDate): Date = date.toDate
 
   private def dateRange(messageKey: String, from: LocalDate, to: LocalDate)(implicit messages: Messages): String =
     if (from isAfter to) {
       throw new IllegalArgumentException(s"From date:$from cannot be after To date:$to")
     } else {
       messages(
         messageKey,
-        HtmlFormatter.htmlNonBroken(Dates.formatDate(from)),
-        HtmlFormatter.htmlNonBroken(Dates.formatDate(to)))
+        HtmlFormatter.htmlNonBroken(createDateFormatForPattern("d MMMM y").format(jodaDate2javaDate(from))),
+        HtmlFormatter.htmlNonBroken(createDateFormatForPattern("d MMMM y").format(jodaDate2javaDate(to)))
+      )
     }
+
+  def formatDate(date: LocalDate)(implicit messages: Messages): String =
+    createDateFormatForPattern("d MMMM y").format(jodaDate2javaDate(date))
+
+  def formatDateAbbrMonth(date: LocalDate)(implicit messages: Messages): String =
```

Comment:
Is the abbreviated method tested? Can find instances of the regular `formatDate` method but not `Abbr`

Reply:
Added test cases for formatDateAbbrMonth in HistoricIncomeCalculationSpec and YourIncomeCalculationViewSpec